js/views/internal_links_support.js

```diff
@@ -179,40 +179,172 @@ var InternalLinksSupport = function(reader) {
 
     this.processLinkElements = function($iframe, spineItem) {
 
-        var epubContentDocument = $iframe[0].contentDocument;
-
-        $('a', epubContentDocument).click(function (clickEvent) {
-            // Check for both href and xlink:href attribute and get value
-            var href;
-            if (clickEvent.currentTarget.attributes["xlink:href"]) {
+        var settings = reader.viewerSettings();
+        if (!settings || typeof settings.epubPopupFootnotes === "undefined")
+        {
+            //defaults
+            settings = new ViewerSettings({});
+        }
+        var epubPopupFootNotesAreEnabled = settings.epubPopupFootnotes;
+        if (epubPopupFootNotesAreEnabled) {
+            var doc = ( $iframe[0].contentWindow || $iframe[0].contentDocument ).document;
+            var $elements = $("*[epub\\:type]", doc);
+            if ($elements.length) {
                 
-                href = clickEvent.currentTarget.attributes["xlink:href"].value;
-            }
-            else {
-                href = clickEvent.currentTarget.attributes["href"].value;
+                var sources = [];
+                var targets = [];
+                
+                $elements.each(function(i) {
+                    
+                    if ($elements[i].localName !== "a"
+                        && $elements[i].localName !== "aside") {
```

Comment:
The limitation to `aside` elements seems very artificial. Is this defined in the EPUB specification? Should the match be based solely on the `epub:type` value?
+ @JayPanoz

Reply:
Yeah so I can tell that at least, iBooks is matching `aside`, `div`, and `p` since they offer two different options to authors as regards hiding the footnote in the normal reading flow (a.k.a. main body): 

- hiding when `aside` is used;
- not hiding it when either `div` or `p` are used. 

See note @ the bottom of https://help.apple.com/itc/booksassetguide/#/itccf8ecf5c8.

You can be sure a large portion of eprdctn knows about that. 

On another amusing note, they expect authors to set the direction inline in case of RTL.

Reply:
Sure, but if I remember correctly, your example used `aside` + `epub:type=footnotes` (plural), with nested `div` + `epub:type=footnote` (singular).  So I modified your example by swapping the `div` and `aside` elements so that it would work correctly in Readium (this code). Perhaps it was a typo at your end? Or was it intentional?

Reply:
Yeah that was intentional → `div + epub:type="footnote"` was there to have the footnote displayed at the end of the XHTML file. If it were `aside + epub:type="footnote"` then iBooks would hide it from the body of the XHTML file. 

Disclaimer: there’s been a lot of a11y issues with popup footnotes in iBooks e.g. screen reader can’t access the contents so others may have used `aside` + nested `div` as well – that + voice over has very weird behaviors in multicol e.g. `alt` not being read for images when they happen to be laid out at the top of a column.

Reply:
And ref to part of their logic if needed: https://gist.github.com/dvschultz/9caba7f9058596451744#file-cfi-js-L1202-L1294

Reply:
+ why only `asides` are hidden and not others: https://github.com/FriendsOfEpub/WillThatBeOverriden/blob/master/ReadingSystems/iBooks/OS%20X/user_stylesheet_flowable.css.tmpl#L232-L234